api/base/caller.go

```diff
@@ -149,15 +146,10 @@ func NewFacadeCaller(caller APICaller, facadeName string, options ...Option) Fac
 // NewFacadeCallerForVersion wraps an APICaller for a given facade
 // name and version.
 func NewFacadeCallerForVersion(caller APICaller, facadeName string, version int, options ...Option) FacadeCaller {
-	// Derive the context from the API caller context if it's available. The
-	// default will be a noop tracer if none is found.
-	tracer, _ := coretrace.TracerFromContext(caller.Context())
```

Comment:
This is a bit unfortunate, as we'll now loose traces from the API caller. I guess we'll just need to wire up the API correctly. It's required to happen. It's just annoying.